src/components/withAuthentication/index.jsx

```diff
@@ -76,6 +79,7 @@ const withAuthentication = (WrappedComponent) => {
 
 withAuthentication.propTypes = {
   location: PropTypes.string.isRequired,
+  providerSlug: PropTypes.string.isRequired,
 };
 
 export default withAuthentication;
```

Comment:
@irfanuddinahmad I was able to convert the 2 HOCs into one HOC. 
```javascript
const withAuthenticationSSO = WrappedComponent => (
  props => (
    <StaticQuery
      query={withSamlQuery}
      render={data => (
        <WrappedComponent providerSlug={data.site.siteMetadata.providerSlug} {...props} />
      )}
    />
  )
);
export default withAuthenticationSSO;
```
However, there was some delay in redirection with an error in `MainContent` component receiving an error object props which was not the instance of `Error` object.